sql/src/main/java/org/apache/druid/sql/calcite/planner/IngestHandler.java

```diff
@@ -106,12 +110,45 @@ protected String operationName()
   @Override
   public void validate()
   {
-    if (ingestNode().getPartitionedBy() == null) {
+    if (ingestNode().getTargetTable() instanceof ExternalDestinationSqlIdentifier) {
```

Comment:
Refactoring comment: This seems slightly messy, that we are interleaving the validation for ingest into datasource along with export. It might be easier to read if the top level validation has following structure:
```java
if (ingestNode.getTargetTable() instanceof ExternalDestinationSqlIdentifier) {
   vallidateExport();
}
else if (ingestNode.getTargetTable() instanceof DataSource(?)) {
  validateIngest();
}
else {
  throw;
}
```